harness/determined/common/api/authentication.py

```diff
@@ -61,6 +61,7 @@ def _init_session(
     ) -> Session:
         session_user = (
             requested_user
+            or util.get_det_username_from_env()
             or self.token_store.get_active_user()
```

Comment:
oh man, I have to think so hard when I get close to this code.

I _think_ that this ordering differs from what you said in your PR comments.  I think this is going to cause the det login user (set in token_store) to never get used, making the `det user login` override you talked about ineffective.

Reply:
I simplified the logic and made it more sane. 

Now both ```DET_USER``` and ```DET_PASS``` must be set to take effect with ```-u``` overriding the environment variables.